frontend/src/components/Modal.js

```diff
@@ -37,18 +43,33 @@ const Modal = ({
       </div>
       <ModalFooter>
         <ButtonGroup>
-          <ModalToggleButton className={cancelButtonCss} data-focus="true" type="button" modalRef={modalRef} closer>
```

Comment:
Probably makes sense for this branch to track the permissions modal one, I see a bunch of duplicated code here (unless I'm confused, which is possible)